src/overlays/actors/ovl_En_Po_Fusen/z_en_po_fusen.c

```diff
@@ -21,34 +29,230 @@ const ActorInit En_Po_Fusen_InitVars = {
+    Collision_AddAC(gCtx, &gCtx->colCheckCtx, &this->collider);
+    return 0;
+}
+
+void EnPoFusen_InitNoFuse(EnPoFusen *this) {
+    this->actor.shape.rot.z = 0;
+    this->actionFunc = EnPoFusen_Idle;
+}
+
+void EnPoFusen_Idle(EnPoFusen *this, GlobalContext *gCtx) {
+    f32 shadowScaleTmp;
+    f32 shadowAlphaTmp;
+    f32 heightOffset;
+    f32 f255 = 255.0f;
+
+    this->actor.world.pos = this->actor.home.pos;
+    this->randScaleChange += 0x190;
+    this->randXZRotChange += 0x5DC;
+    this->randBaseRotChange += 0x9C4;
+    heightOffset = Math_Sins(this->randScaleChange);
+    heightOffset = 50.0f * heightOffset;
+    this->actor.shape.rot.y += this->randYRotChange;
+    this->actor.world.pos.y += heightOffset;
+    this->actor.shape.rot.z =  (Math_Sins(this->randBaseRotChange) * 910.0f);
+    if ((this->randScaleChange < 0x4000) && (this->randScaleChange >= -0x3FFF)) {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 0x38E, 0x14, 0xBB8, 0x64);
+    } else {
+        Math_SmoothScaleMaxMinS( &this->limb9Rot, 0x71C, 0x8, 0xBB8, 0x64);
+    }
+    this->avgBaseRotation = this->limb9Rot * 3;
+    this->limb3Rot  = (Math_Sins(this->randBaseRotChange + 0x38E3) * this->avgBaseRotation);
+    this->limb46Rot = (Math_Sins(this->randBaseRotChange)          * this->avgBaseRotation);
+    this->limb57Rot = (Math_Sins(this->randBaseRotChange - 0x38E3) * this->avgBaseRotation);
+    this->limb8Rot  = (Math_Sins(this->randBaseRotChange - 0x71C6) * this->avgBaseRotation);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/EnPoFusen_Destroy.asm")
+    shadowScaleTmp = ((1.0f - Math_Sins(this->randScaleChange)) * 10.0f) + 50.0f;
+    shadowAlphaTmp = ((1.0f - Math_Sins(this->randScaleChange)) * 75.0f) + 100.0f;
+    this->actor.shape.shadowScale = shadowScaleTmp;
+    this->actor.shape.shadowAlpha = (shadowAlphaTmp > f255) ? (u8) f255 : (u8) shadowAlphaTmp;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC1450.asm")
+void EnPoFusen_IncrementMalonPop(EnPoFusen *this) {
+    Actor* parent = this->actor.parent;
+    EnMa4* romani;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC14A4.asm")
+    if ((parent != 0) && (parent->id == ACTOR_EN_MA4)) { 
+        romani = (EnMa4*) parent;
+        romani->unk338 += 1;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC1574.asm")
+    this->actor.draw = NULL;
+    this->actionFunc = EnPoFusen_Pop;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/func_80AC158C.asm")
+void EnPoFusen_Pop(EnPoFusen *this, GlobalContext *gCtx) {
+    Actor_Spawn(&gCtx->actorCtx, gCtx, ACTOR_EN_CLEAR_TAG,
+       this->actor.world.pos.x, this->actor.world.pos.y + 20.0f, this->actor.world.pos.z, 
+       255, 255, 200, 2);
+    func_800B8EC8(&this->actor, 0x180E); // NA_SE_IT_BOMB_EXPLOSION sfx
```

Comment:
should we introduce a define here @Kenix3 ?

Reply:
on top of that, should we just make actor_spawn a single line?

Reply:
I won't require it for this PR, but I'd be fine with starting up a `sfx.h` like in OOT

Reply:
Yea thats fine